AutoRest/AutoRest.Core/ClientModel/Method.cs

```diff
@@ -59,6 +60,22 @@ public Method()
         public List<Parameter> Parameters { get; private set; }
 
         /// <summary>
+        /// Gets the list of Parameter Expansions
+        /// </summary>
+        public Dictionary<string, Dictionary<Property, Parameter>> ParameterExpansions { get; private set; }
```

Comment:
To make it more generic and support parameter groups as well primitive to object mapping can we do the following:
1. Rename this to ParameterMappings
2. Add new type to ClientModel called `ParameterMapping` with the following properties
   - `Parameter InputParameter {get; set;}`
   - `string InputParameterProperty {get; set;}` (either null, which will map the parameter itself, or dot separated string to match properties)
   - `Parameter OutputParameter {get; set;}`
   - `string OutputParameterProperty {get; set;}` (either null, which will map the parameter itself, or dot separated string to match properties)